libcoveoc4ids/additional_checks.py

```diff
@@ -0,0 +1,91 @@
+import gettext
+import re
+
+from libcoveoc4ids.additional_check import AdditionalCheck
+
+_ = gettext.gettext
+
+
+class CurrencyCheck(AdditionalCheck):
+    def process(self, data, flat_data):
+        amount_path = re.compile(".*amount$")
+
+        expected_currency_paths = []
+        all_paths = []
+        missing = []
+
+        for path in flat_data.keys():
+            if amount_path.match(path):
+                expected_currency_paths.append(path[: -len("amount")] + "currency")
+
+            all_paths.append(path)
+
+        for expected in expected_currency_paths:
+            if expected not in all_paths:
+                missing.append(expected[: -len("currency")])
+
+        if len(missing) == 0:
+            return True
+
+        return self.result(
+            "missing-currency",
+            _("There are %s values without a currency, Currencies should be published for all values" % len(missing)),
```

Comment:
The `_(` should wrap the static string (with the `%s`), rather than the composed string.

Also, I've been mostly using the new style `.format(` instead of the old style `%s`, so it could be worth switching that for consistency.

Reply:
Thanks - updated

Reply:
While I prefer `.format()` to `%` in general, there are two exceptions:

1. logging. If you do `logger.info("string %s to format", arg)` (lazy evaluation style) then the string is only formatted if the logger needs to output that message (if the logging level is warning, for example, then there will be no formatting). In our case, it's unlikely the case that logging will affect performance, but the above is nonetheless the preferred style.
1. i18n. Django (and other gettext-based systems) prefer named-string interpolation using `%`-formatting, like `%(number)d` https://docs.djangoproject.com/en/3.1/topics/i18n/translation/#standard-translation

I'll add something here: https://ocp-software-handbook.readthedocs.io/en/latest/python/code.html

Reply:
> While I prefer `.format()` to `%` in general, there are two exceptions:
> 
> 1. logging. If you do `logger.info("string %s to format", arg)` (lazy evaluation style) then the string is only formatted if the logger needs to output that message (if the logging level is warning, for example, then there will be no formatting). In our case, it's unlikely the case that logging will affect performance, but the above is nonetheless the preferred style.
> 2. i18n. Django (and other gettext-based systems) prefer named-string interpolation using `%`-formatting, like `%(number)d` https://docs.djangoproject.com/en/3.1/topics/i18n/translation/#standard-translation
> 
> I'll add something here: https://ocp-software-handbook.readthedocs.io/en/latest/python/code.html

No problem, have now updated the draft PR